changes/2809.bugfix.rst

```diff
@@ -0,0 +1 @@
+When Toga creates asyncio Tasks for widget and app handlers, the garbage collector will no longer prematurely destroy these tasks in the unlikely occasions it would occur previously.
```

Comment:
```suggestion
Asynchronous tasks created by Toga are now protected from garbage collection while they are running. This could lead to asynchronous tasks terminating unexpectedly with an error under some conditions.
```